localstack/services/opensearch/cluster.py

```diff
@@ -118,6 +121,104 @@ def build_cluster_run_command(cluster_bin: str, settings: CommandSettings) -> Li
     return [cluster_bin] + cmd_settings
 
 
+class CustomEndpoint:
+    """
+    Encapsulates a custom endpoint (combines CustomEndpoint and CustomEndpointEnabled within the DomainEndpointOptions
+    of the cluster, i.e. combines two fields from the AWS OpenSearch service model).
+    """
+
+    enabled: bool
+    endpoint: str
+
+    def __init__(self, enabled: bool, endpoint: str) -> None:
+        """
+        :param enabled: true if the custom endpoint is enabled (refers to DomainEndpointOptions#CustomEndpointEnabled)
+        :param endpoint: defines the endpoint (i.e. the URL - refers to DomainEndpointOptions#CustomEndpoint)
+        """
+        self.enabled = enabled
+        self.endpoint = endpoint
+
+        if self.endpoint:
+            self.url = urlparse(endpoint)
+        else:
+            self.url = None
+
+
+def register_cluster(
+    host: str, path: str, forward_url: str, custom_endpoint: CustomEndpoint
+) -> List[Rule]:
+    """
+    Registers routes for a cluster at the edge router.
+    Depending on which endpoint strategy is employed, different routes are registered.
+    This method is tightly coupled with build_cluster_endpoint(). It prepares the necessary URL data already according
+    to the endpoint strategies. It also has an overview of the different endpoint strategies and their url schemas.
+
+    :param host: hostname of the inbound address without protocol or port
+    :param path: path of the inbound address
+    :param forward_url: whole address for outgoing traffic (including the protocol)
+    :param custom_endpoint: Object that stores a custom address and if its enabled.
+            If a custom_endpoint is set AND enabled, the specified address takes precedence
+            over any strategy currently active, and overwrites any host/path combination.
+
+    :return: A list of generated router rules. Primarily a reference to them for cleanup on shutdown.
```

Comment:
Some notes on the docs here:
- "schema" is confusing in this context, because a URL "scheme" is defined to be a defined format specification (like "http", "https", or "ftp") and is always the first part of a URL.
- `build_cluster_endpoint` is not in the same module, you should mention that it's in the cluster_manager module.
- This function doesn't prepare URLs but routing rules.
Here's a suggestion:

```suggestion
    Registers routes for a cluster at the edge router.
    Depending on which endpoint strategy is employed, and if a custom endpoint is enabled, different routes are
    registered.
    This method is tightly coupled with `cluster_manager.build_cluster_endpoint`, which already creates the
    endpoint URL according to the configuration used here.
    
    :param host: hostname of the inbound address without scheme or port
    :param path: path of the inbound address
    :param forward_url: whole address for outgoing traffic (including the protocol)
    :param custom_endpoint: Object that stores a custom address and if its enabled.
            If a custom_endpoint is set AND enabled, the specified address takes precedence
            over any strategy currently active, and overwrites any host/path combination.

    :return: a list of generated router rules, which can be used for removal
```